src/execution/window_executor.cpp

```diff
@@ -207,19 +207,24 @@ static idx_t FindTypedRangeBound(const WindowInputColumn &over, const idx_t orde
 	WindowColumnIterator<T> begin(over, order_begin);
 	WindowColumnIterator<T> end(over, order_end);
 
-	if (order_begin < prev.start && prev.start < order_end) {
-		const auto first = over.GetCell<T>(prev.start);
-		if (!comp(val, first)) {
-			//	prev.first <= val, so we can start further forward
-			begin += (prev.start - order_begin);
+	//	Try to reuse the previous bounds to restrict the search.
+	//	This is only valid if the previous bounds were non-empty
+	//	Only inject the comparisons if the previous bounds are a strict subset.
+	if (prev.start < prev.end) {
```

Comment:
Why is this condition false, if we randomize the row order? And we could have an early-out in that case.

Reply:
The randomisation just changes which peer row comes first and that can change the result - even making it correct if the min row happens to come first.

Reply:
> And we could have an early-out in that case 

This is a performance optimisation - the test is for the _previous_ range points, not the current ones. We always have to check the current ones.

Reply:
Okay, thanks! What tipped me off is that we're comparing `start < end` of the same `prev`.